File: user-service/src/main/java/com/stackroute/userservice/model/UserProfile.java
package com.stackroute.userservice.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Document
public class UserProfile {
    @Id
    private String email;
    private String name;
    private Date creationDate;
    private String age;
    private String place;
    private String imageUrl;
    private String planType;
    private String physicalLevel;
    private String mentalLevel;
    private String dietLevel;
    private double physicalScore;
    private double mentalScore;
    private double dietScore;
    List<Users> followers;
    List<Users> following;
    List<Mentor> mentorList;
    Plan plan;
    List<Day> days;
    List<Appointment> appointments;
}

File: wholesome-care-frontend/src/app/dashboard/UserProfile.ts
export class UserProfile {
  email:string;
  name:string;
  creationDate:string;
  age:string;
  place:string;
  imageUrl:string;
  planType:string;
  physicalLevel:string;
  mentalLevel:string;
  dietLevel:string;
  followers:Array<any>;
  following:Array<any>;
  mentorList:Array<any>;
  plan:{activities:Array<any>, activitiesDone:Array<any>, diet_plan:any};
  days:Array<{localDate:string, plan:{activities:Array<any>, activitiesDone:Array<any>, diet_plan:any}}>;
  constructor(email:string, name:string, creationDate:string, age:string, place:string, imageUrl:string, planType:string,
    physicalLevel:string, mentalLevel:string, dietLevel:string, followers:Array<any>, following:Array<any>, mentorList:Array<any>,
    plan:{activities:Array<any>, activitiesDone:Array<any>, diet_plan:any}, days:Array<{localDate:string, plan:{activities:Array<any>, activitiesDone:Array<any>, diet_plan:any}}>){
      this.email=email;
      this.name=name;
      this.creationDate=creationDate;
      this.age=age;
      this.place=place;
      this.imageUrl=imageUrl;
      this.planType=planType;
      this.physicalLevel=physicalLevel;
      this.mentalLevel=mentalLevel;
      this.dietLevel=dietLevel;
      this.followers=followers;
      this.following=following;
      this.mentorList=mentorList;
      this.plan=plan;
      this.days=days;
  }
}
